viadot/sources/spectrum.py

```diff
@@ -0,0 +1,166 @@
+    """
+
+    def __init__(
+        self,
+        profile_name: str = None,
+        aws_access_key_id: str = None,
+        aws_secret_access_key: str = None,
+    ):
+        if profile_name:
+            self.session = boto3.session.Session(profile_name=profile_name)
+        elif aws_access_key_id and aws_secret_access_key:
+            self.session = boto3.session.Session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+            )
+        else:
+            self.session = boto3.session.Session()
+
+    def ls(
+        self, database: str = None, name_contains: str = None, search_text: str = None
+    ) -> List[str]:
+        """
+        Returns a list of tables in a Glue database.
+
+        Args:
+            database (str, optional): Database name.
+            name_contains (str, optional): Select by a specific string on table name.
+            search_text (str, optional): Select only tables with the given string in table's properties.
+        """
+        df = wr.catalog.tables(
+            boto3_session=self.session,
+            database=database,
+            name_contains=name_contains,
+            search_text=search_text,
+        )
+
+        return df
+
+    def exists(self, database: str, table: str) -> bool:
+        """
+        Check if a table exists in Glue database.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+
+        Returns:
+            bool: Whether the paths exists.
+        """
+        return wr.s3.does_table_exist(
+            boto3_session=self.session, database=database, table=table
+        )
+
+    def rm(
+        self,
+        database: str,
+        table: str,
+    ):
+        """
+        Deletes table from Glue database.
+
+        Args:
+            path (str): Path to a folder.
+            database (str): AWS Glue  catalog database name.
+            table (str): AWS Glue catalog table name.
+        """
+        table_location = wr.catalog.get_table_location(
+            boto3_session=self.session, database=database, table=table
+        )
+        wr.catalog.delete_table_if_exists(database=database, table=table)
+
+        wr.s3.delete_objects(path=table_location)
+
+    def from_df(
+        self,
+        df: pd.DataFrame,
+        to_path: str,
+        database: str,
+        table: str,
+        extention: str = ".parquet",
+        if_exists: Literal["overwrite", "append"] = "overwrite",
+        partition_cols: List[str] = None,
+        index: bool = False,
+        compression: str = None,
+        sep: str = ",",
+        description: str = "test",
+        columns_comments: Dict[str, str] = None,
+    ):
+        """
+        Upload a pandas `DataFrame` to a csv or parquet file.
+
+        Args:
+            df (pd.DataFrame): Pandas DataFrame
+            to_path (str): Path to a S3 folder. Defaults to None.
```

Comment:
"a s3 folder" is not informative, I think "the s3 folder where the table will be located" would be more useful.

Reply:
Updated as per [🎨 updated docstrings](https://github.com/dyvenia/viadot/pull/579/commits/d0cf334bab18faf483e465d88ab96b537e31b9ed)